shim.go

```diff
@@ -100,6 +110,13 @@ func newShimConfig(config PodConfig) interface{} {
 			return err
 		}
 		return ccConfig
+	case KataShimType:
+		var kataConfig KataShimConfig
```

Comment:
Same thing I did for the proxy, I think we should not create different config structures for each implementation. We could simplify everything into a `ShimConfig` structure because I don't see any reason why we would need different configs (we expect a shim to be kind of the same thing). This can be done as a follow up PR I guess.